bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/ExperienceFragmentImpl.java

```diff
@@ -0,0 +1,221 @@
+            String xfRootPath = getXfRoot(fragmentPath, currentPageRootPath);
+            if (!StringUtils.isEmpty(currentPageRootPath) && !StringUtils.isEmpty(xfRootPath)) {
+                String xfRelativePath = StringUtils.substring(fragmentPath, xfRootPath.length());
+                String localizedXfRootPath = StringUtils.replace(currentPageRootPath, CONTENT_ROOT, EXPERIENCE_FRAGMENTS_ROOT, 1);
+                localizedFragmentPath = StringUtils.join(localizedXfRootPath, xfRelativePath, JCR_CONTENT_ROOT);
+            }
+        }
+        String xfContentPath = StringUtils.join(fragmentPath, JCR_CONTENT_ROOT);
+        if (!resourceExists(localizedFragmentPath) && resourceExists(xfContentPath)) {
+            localizedFragmentPath = xfContentPath;
+        }
+    }
+
+    @Override
+    public String getFragmentPath() {
+        return fragmentPath;
+    }
+
+    @Override
+    public String getLocalizedFragmentPath() {
+        return localizedFragmentPath;
+    }
+
+    @NotNull
+    @Override public String getExportedType() {
+        return request.getResource().getResourceType();
+    }
+
+    /**
+     * Returns the localization root of the resource defined at the given path
+     *
+     * Use case                                  | Path                                 | Root
+     * ----------------------------------------  |--------------------------------------|------------------
+     * 1. No localization                        | /content/mysite/mypage               | null
+     * 2. Language localization                  | /content/mysite/en/mypage            | /content/mysite/en
+     * 3. Country-language localization          | /content/mysite/us/en/mypage         | /content/mysite/us/en
+     * 4. Country-language localization (variant)| /content/us/mysite/en/mypage         | /content/us/mysite/en
+     * 5. Blueprint                              | /content/mysite/blueprint/mypage     | /content/us/mysite/blueprint
+     * 4. Livecopy                               | /content/mysite/livecopy/mypage      | /content/us/mysite/livecopy
+     *
+     * @param path the resource path.
+     * @return the localization root of the resource at the given path if it exists, {@code null} otherwise.
+     */
+    private String getRoot(String path) {
+        String root = null;
+        if (!StringUtils.isEmpty(path)) {
+            Resource resource = resolver.getResource(path);
+            if (resource != null) {
+                Page rootPage = languageManager.getLanguageRoot(resource);
+                if (rootPage != null) {
+                    // language root
+                    root = rootPage.getPath();
+                }
+                if (StringUtils.isEmpty(root)) {
+                    // no language root
+                    try {
+                        if (relationshipManager.isSource(resource)) {
+                            // the resource is a blueprint
```

Comment:
you could replace comments with dedicated methods e.g.
`if (relationshipManager.isSource(resource))` -> `if(isBlueprintResource(resource))`
`(relationshipManager.hasLiveRelationship(resource)` -> `if(isLiveCopyResource(resource))`

Reply:
I would say that it's clear enough the was it is now.